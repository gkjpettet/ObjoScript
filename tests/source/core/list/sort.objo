class Person {
constructor(name, age) { 
_name = name
_age = age 

}
>(other) { return age > other.age }
<(other) { return age < other.age }
<=(other) { return age <= other.age }
>=(other) { return age >= other.age }
==(other) { return age == other.age }

age {return _age}

toString() { return _name }
}

var garry = Person("Garry", 41)
var aoife = Person("Aoife", 10)
var carole = Person("Carole", 68)
var fiona = Person("Fiona", 41)
var gaga = Person("Gaga", 74)

var people = [gaga, garry, fiona, aoife, carole]

people.sort

// Sort by age ascending.
foreach person in people { System.print(person.toString) }