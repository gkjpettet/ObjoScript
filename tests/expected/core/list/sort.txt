Aoife
Fiona
Garry
Carole
Gaga